vignettes/programming_strategy.Rmd

```diff
@@ -392,6 +392,91 @@ Make sure to align columns as well. This ensures quick code readability.
 
 If a unit test fails, it must provide a helpful error message, e.g., if the output dataset is not as expected, the observations, variables, and values which differ should be provided. See also test helper function (link to be added).
 
+# Deprecation
+
+As admiral is still evolving, functions or parameters may need to be removed or replaced with more 
+efficient options from one release to another. In such cases, the relevant function or parameter 
+must be marked as deprecated. 
+
+## Documentation
+
+If a function or parameter is removed, the documentation must be updated to indicate the function 
+or the parameter is now deprecated and which new function/parameter should be used instead.
+
+The documentation will be updated at:
+
++ the main title (??? or another section???) level for a function, 
+
+```
+#' Title of the function
+#'*Deprecated*, please use `new_fun()` instead.
+#'
+#' @description
+#' description of the function in plain english
+``` 
+
++ the `@param` level for a parameter.
+
+```
+@param old_param *Deprecated*, please use `new_param` instead.
+```
+
+## Handling of warning and error
+
+When a function or parameter is deprecated, the function must be updated to issue an error or a 
+warning to inform the user.
+
+### Function
+
+If the deprecated function still exist in the package besides the new function, a **warning** must be
+issued. If if it has been removed, an **error** must be generated.
```

Comment:
I was thinking we would keep the deprecated function as:
```
deprecated_fns<-function(x) {
  deprecate_stop("x.y.z",  "deprecated_fns", "new_fns")
}
```

So that the user knows the function is now removed and what to use instead.